src/lib/mina_transition/external_transition.ml

```diff
@@ -392,7 +436,7 @@ module Validation = struct
        , 'staged_ledger_diff
        , 'protocol_versions )
        with_transition =
-    external_transition State_hash.With_state_hashes.t
+    Block.with_hash
```

Comment:
I would prefer to move this validation logic elsewhere, and to patch it up as I did in my original PR. The old validation implementation is actually wrong in a number of ways, which is why I had re-implemented a new validation module and ported over the old validation code to it.